templates/report.js

```diff
@@ -2,4 +2,75 @@ angular.module('app', [])
   .controller('ReportController', function($scope, $locale) {
     $scope.results = results;
     $scope.jsonData = JSON.stringify(results, null, 4);
+
+    function countKPIs() {
+      var metrics = {
+        totalSuites: 0,
+        passedSuites: 0,
+        totalSpecs: 0,
+        passedSpecs: 0,
+        totalExp: 0,
+        passedExp: 0,
+        timeMin: 0,
+        timeSec: 0
+      };
+
+      function traverse(node) {
+        console.log(node.type + ' - ' + node.id );
+
+        var isPass = true;
+        if (node.children) {
+          for (var i = 0; i <  node.children.length; i++) {
+            isPass &= traverse(node.children[i]);
+          }
+        }
+
+        switch (node.type) {
+          case 'suite':
+            // we are counting suites at ALL levels.
+            metrics.totalSuites ++;
+            if (isPass) {
+              metrics.passedSuites ++;
+            }
+            break;
+
+          case 'spec':
+            var spec = node;
+            isPass = (spec.failedExpectations.length == 0);
+            metrics.totalExp += (spec.failedExpectations.length + spec.passedExpectations.length);
+            metrics.passedExp += spec.passedExpectations.length;
+
+            metrics.totalSpecs ++;
+            if (isPass) {
+              metrics.passedSpecs ++;
+            }
+            break;
+
+          default:
+            break;
+        }
+
+        return isPass;
+      }
+
+      traverse(results);
+
+      results.startTime = results.startTime[0] + (results.startTime[1] * (1e-9));
```

Comment:
1. we just need seconds resolution. we might as well throw away the nanoseconds.. 
2. it is not a good idea to overwrite a field like this. use a separate variable if need to.

Reply:
what I was saying is, 
consider [10, 384474] - [5, 928383] = 4 seconds (it is 4.xx)
consider [10, 839934] - [5, 564949] = 5 seconds (it is 5.xx)